runhouse/main.py

```diff
@@ -666,6 +351,182 @@ def cluster_list(
+        return
+
+    elif cluster_name:
+
+        try:
+            current_cluster = cluster(name=cluster_name)
+        except ValueError:
+            console.print("Failed to load status for cluster.")
+            return
+
+        if not current_cluster.is_up():
+            print_bring_cluster_up_msg(
+                cluster_name=cluster_name, msg_prefix="Can't get live cluster logs"
+            )
+            bring_logs_from_den = True
+
+        try:
+            if current_cluster._http_client:
+                current_cluster._http_client.check_server()
+        except requests.exceptions.ConnectionError:
+            print_bring_cluster_up_msg(
+                cluster_name=cluster_name, msg_prefix="Can't get live cluster logs"
+            )
+            bring_logs_from_den = True
+
+    elif not cluster_or_local:
+        # todo: make func as the print_bring_cluster_up_msg
+        console.print(
+            "Can't get live cluster logs: \N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}. "
+            "Start it with `runhouse restart` or specify a remote "
+            "cluster to poll with `runhouse status <cluster_name>`."
+        )
+        bring_logs_from_den = True
+
+    else:
+        # we are inside the cluster
+        current_cluster = cluster_or_local  # cluster_or_local = rh.here
+
+    try:
+        if bring_logs_from_den:
+            console.print("Getting the last chunk of logs saved in Den.")
+            get_logs_den_resp = current_cluster._get_saved_logs_from_den()
+            if get_logs_den_resp.status_code != 200:
+                console.print("Failed to get logs from den.")
+                return
+            logs = "\n".join(get_logs_den_resp.json().get("data").get("logs_text"))
+        else:
+            logs = current_cluster.logs(tail)
+
+    except requests.exceptions.ConnectionError:
+        console.print("Can't get live cluster logs, ")
+        console.print(
+            "\N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}"
+        )
+        return
+
+    console.print(f"[reset][italic cyan]{logs}")
+
+
+@cluster_app.command("up")
+def cluster_up(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to bring up.",
+    )
+):
+    """Bring up the cluster if it is not up. No-op if cluster is already up.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+        .. code-block:: console:
+
+            $ runhouse cluster up /sashab/rh-basic-cpu
+
+    """
+    try:
+        current_cluster = rh.cluster(name=cluster_name)
+        current_cluster.up_if_not()
+    except ValueError as e:
+        if str(e) == f"Resource {cluster_name} not found":
+            console.print("Cluster is not saved in den.")
+        else:
+            console.print(f"Failed to bring up the cluster: {e}")
```

Comment:
what other value error could be raised here?

Reply:
a value error that is not raised by runhouse. This is an edge case.. I'm on the fence deciding if it will be common enough to have this fork in the code. wdyt?

Reply:
is it something you've seen come up? if not i think we can remove it